Source/DiabloUI/diabloui.cpp

```diff
@@ -621,19 +621,27 @@ void UiDestroy()
 	UnloadUiGFX();
 }
 
-bool UiValidPlayerName(const char *name)
+bool UiValidPlayerName(string_view name)
 {
-	if (strlen(name) == 0)
+	if (name.empty())
 		return false;
 
-	if (strpbrk(name, ",<>%&\\\"?*#/:") != nullptr || strpbrk(name, " ") != nullptr)
+	// Currently only allow saving PLR_NAME_LEN bytes as a player name, so if the name is too long we'd have to truncate it.
+	// That said the input buffer is only 16 bytes long...
+	if (name.size() > PLR_NAME_LEN)
 		return false;
 
-	for (BYTE *letter = (BYTE *)name; *letter != '\0'; letter++)
-		if (*letter < 0x20 || (*letter > 0x7E && *letter < 0xC0))
+	if (name.find_first_of(",<>%&\\\"?*#/: ") != name.npos)
+		return false;
+
+	constexpr auto isAsciiControlCharacter = [](char character) -> bool {
+		return (character >= 0x00 && character < 0x20) || character == 0x7F;
+	};
+	for (char letter : name)
+		if (isAsciiControlCharacter(letter) || !IsLeadUtf8CodeUnit(letter))

```

Comment:
Looks like this checks that *every* byte is a lead code unit, so would break support for non-ASCII names?

Reply:
In multiplayer non-ASCII names where not permitted.

Reply:
Yeah, the original code looks like it was only intended to support basic latin but it did this by looking for the presence of trailing code points specifically. Or it was assuming an ascii encoding and blocking half the extended range for some reason?

I didn't change it here as I didn't want to change behaviour but this could be simplified to a single check `isBasicLatin`.

Reply:
Added a commit which changes this block of code to just check for basic latin characters. I think this conveys intent better given devilutionX uses utf8 strings.